Explain gas implications of constant and immutable variables. (#9510)
ethereum/solidity: docs/contracts/constant-state-variables.rst

The compiler does not reserve a storage slot for these variables, and every occurrence is
replaced by the respective value.

Compared to regular state variables, the gas costs of constant and immutable variables

> are much lower. For a constant variables, the expression assigned to it is copied to

Reviewer: ```suggestion
are much lower. For constant variables, the expression assigned to it is copied to
```

> are much lower. For a constant variables, the expression assigned to it is copied to

Reviewer 2: ```suggestion
are much lower. For a constant variable, the expression assigned to it is copied to
```

all the places where it is accessed and also re-evaluated each time. This allows for local
optimizations. Immutable variables are evaluated once at construction time and their value

> is copied to all the places where they are accessed. For this, 32 bytes are reserved

Reviewer: ```suggestion
is copied to all the places where they are accessed. 32 bytes are reserved
```

Author: I try to avoid starting a sentence with a number, so I did a different change.

> for the value, even if it is smaller. Because of that, constant values could be cheaper

Reviewer: ```suggestion
to hold the value of an immutable variable, even if its assigned value is smaller. Due to this, constant values could be cheaper
```

> than immutable value.

Reviewer: ```suggestion
than immutable values.
```

> than immutable value.

Reviewer 2: ```suggestion
than immutable values.
```
